scripts/codegen.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"log"
+	"os"
+	"text/template"
+
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+var (
+	rawTpl = `/*
+DO NOT EDIT
+Generated by codegen.go
+https://github.com/googleapis/go-genproto/blob/0135a39c27378c1b903c75204eff61a060be5eb7/googleapis/monitoring/v3/common.pb.go
+*/
+package tsplot
+{{range $name, $value := .Aligners}}
+func (mq *MetricQuery) Set_{{$name}}() {
+	*mq.aggregation = append(*mq.aggregation, withPerSeriesAligner({{$value}}))
+}
+{{end -}}
+{{range $name, $value := .Reducers}}
+func (mq *MetricQuery) Set_{{$name}}() {
+	*mq.aggregation = append(*mq.aggregation, withCrossSeriesReducer({{$value}}))
+}
+{{end}}
+`
+)
+
+func main() {
+	f, err := os.OpenFile("../tsplot/set_aggregation_opts.go", os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0644)
```

Comment:
todo: 
1. parameterize output.
2. add target to makefile to generate
3. CI check to ensure file is not modified.